travis/script/format/check_java.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+shopt -s globstar
+
+project_root_dir=$1
+formatter_jar_name=formatter.jar
+
+function download_formatter() {
+  local jar_name=$1
+  local url="https://repo.maven.apache.org/maven2/com/google/googlejavaformat/google-java-format/1.6/google-java-format-1.6-all-deps.jar"
+  echo "Downloading formatter from $url"
+  curl -Lo formatter.jar "$url"
+}
+
+function check_formatting() {
+  local jar_name=$1
+
+  echo 'Checking format'
+  java -jar "$jar_name" --dry-run --set-exit-if-changed **/*.java
```

Comment:
If just check changed files is enough you could maybe use the file list of `git diff master --name-only`.